aws-opentelemetry-distro/src/amazon/opentelemetry/distro/exporter/otlp/aws/metrics/otlp_aws_emf_exporter.py

```diff
@@ -0,0 +1,482 @@
+        "By": "Bytes",
+        "KiBy": "Kilobytes",
+        "MiBy": "Megabytes",
+        "GiBy": "Gigabytes",
+        "TiBy": "Terabytes",
+        "Bi": "Bits",
+        "KiBi": "Kilobits",
+        "MiBi": "Megabits",
+        "GiBi": "Gigabits",
+        "TiBi": "Terabits",
+        "%": "Percent",
+        "1": "Count",
+        "{count}": "Count",
+    }
+
+    def __init__(
+        self,
+        namespace: str = "default",
+        log_group_name: str = None,
+        log_stream_name: Optional[str] = None,
+        aws_region: Optional[str] = None,
+        preferred_temporality: Optional[Dict[type, AggregationTemporality]] = None,
+        **kwargs,
+    ):
+        """
+        Initialize the CloudWatch EMF exporter.
+
+        Args:
+            namespace: CloudWatch namespace for metrics
+            log_group_name: CloudWatch log group name
+            log_stream_name: CloudWatch log stream name (auto-generated if None)
+            aws_region: AWS region (auto-detected if None)
+            preferred_temporality: Optional dictionary mapping instrument types to aggregation temporality
+            **kwargs: Additional arguments passed to botocore client
+        """
+        super().__init__(preferred_temporality)
+
+        self.namespace = namespace
+        self.log_group_name = log_group_name
+        self.log_stream_name = log_stream_name or self._generate_log_stream_name()
+
+        # Initialize CloudWatch Logs client using botocore
+        # If aws_region is not provided, botocore will check environment variables AWS_REGION or AWS_DEFAULT_REGION
+        if aws_region is None:
+            aws_region = os.environ.get("AWS_REGION") or os.environ.get("AWS_DEFAULT_REGION")
+
+        session = botocore.session.Session()
+        self.logs_client = session.create_client("logs", region_name=aws_region, **kwargs)
+
+        # Ensure log group exists
+        self._ensure_log_group_exists()
+
+    def _generate_log_stream_name(self) -> str:
+        """Generate a unique log stream name."""
+
+        unique_id = str(uuid.uuid4())[:8]
+        return f"otel-python-{unique_id}"
+
+    def _ensure_log_group_exists(self):
+        """Ensure the log group exists, create if it doesn't."""
+        try:
+            self.logs_client.describe_log_groups(logGroupNamePrefix=self.log_group_name, limit=1)
+        except ClientError:
+            try:
+                self.logs_client.create_log_group(logGroupName=self.log_group_name)
+                logger.info("Created log group: %s", self.log_group_name)
+            except ClientError as error:
+                logger.error("Failed to create log group %s : %s", self.log_group_name, error)
+                raise
+
+    def _get_metric_name(self, record) -> Optional[str]:
+        """Get the metric name from the metric record or data point."""
+        # For metrics in MetricsData format
+        if hasattr(record, "name") and record.name.strip():
+            return record.name
```

Comment:
From what I saw in the full EMF Exporter implementation, `record.name` is never set. `record.instrument.name` is always set in `_create_metric_record`. So this `if statement` is redundant.